CHANGELOG.md

```diff
@@ -0,0 +1,8 @@
+# Change Log
```

Comment:
It is up to the individual projects, what they prefer. What I do with PyIceberg is use the GitHub milestones to keep track of what's being included in a version. You can link both issues and PRs to the milestone, and that works well for me.

Reply:
Github ecosystem like dependabot will read `changelog.md` to provide more information for users, for example: 
![image](https://github.com/apache/iceberg-rust/assets/5351546/0f6db3d6-246e-446a-8052-3a15646c9f5c)

And I plan to utilize GitHub's built-in release note generator, which is based on pull requests, to assist us in creating the changelog. This should result in a minimal maintenance burden.